host_modules/systemd_service.py

```diff
@@ -48,3 +50,19 @@ def stop_service(self, service):
         if result.returncode:
             msg = result.stderr.decode()
         return result.returncode, msg
+
+    @host_service.method(host_service.bus_name(MOD_NAME), in_signature='s', out_signature='is')
+    def execute_reboot(self, rebootmethod):
+        if rebootmethod in REBOOTMETHOD_COLD_BOOT_VALUES:
+            cmd = ['/usr/local/bin/reboot']
+        elif rebootmethod in REBOOTMETHOD_HALT_BOOT_VALUES:
+            cmd = ['/usr/local/bin/reboot','-p']
```

Comment:
>-p [](http://example.com/codeflow?start=44&length=2)

I could not find the implementation of `-p` in sonic repos.

Reply:
@qiluo-msft, this change is upcoming in sonic-utilties which is work-in-progress. If the existing image does not have "-p", command will fail. Thank you!

Reply:
So this PR could not be merged until the depended PR in other repos merged first.

Reply:
depending on https://github.com/sonic-net/sonic-utilities/pull/3566

Reply:
@qiluo-msft "reboot supports '-p' option only on DPU's. I have made a hack to the changes and verified the '-p' option on the non-smartswitch and its successfully running. Pasting output of command JFYI"

`$ sudo reboot -p
/host: 112.7 MiB (118222848 bytes) trimmed on /dev/sda3
INFO: Locking /tmp/mlxfwmanager-lock from mlnx-fw-upgrade.sh
INFO: Locked /tmp/mlxfwmanager-lock (10) from mlnx-fw-upgrade.sh
INFO: Spectrum ASIC successfully detected at /dev/mst/mt52100_pci_cr0

INFO: firmware is up to date

INFO: Unlocking /tmp/mlxfwmanager-lock (10) from mlnx-fw-upgrade.sh
Watchdog armed for 180 seconds
 pre-shutdown steps are completed
$`

Reply:
Also, for switches without "-p" support will throw error as below:

```
$ sudo reboot -p
/usr/local/bin/reboot: illegal option -- p
Usage /usr/local/bin/reboot [options]
    Request rebooting the device. Invoke platform-specific tool when available.
    This script will shutdown syncd before rebooting.
 
    Available options:
        -h, -? : getting this help
```